lib/WebSocketServer.js

```diff
@@ -44,67 +56,52 @@ function WebSocketServer (options, callback) {
     backlog: null // use default (511 as implemented in net.js)
   }, options);
 
-  if (!isDefinedAndNonNull(options, 'port') && !isDefinedAndNonNull(options, 'server') && !options.noServer) {
-    throw new TypeError('`port` or a `server` must be provided');
+  if (options.port == null && !options.server && !options.noServer) {
+    throw new TypeError('missing or invalid options');
   }
 
-  if (isDefinedAndNonNull(options, 'port')) {
+  if (options.port != null) {
     this._server = http.createServer((req, res) => {
-      var body = http.STATUS_CODES[426];
+      const body = http.STATUS_CODES[426];
+
       res.writeHead(426, {
         'Content-Length': body.length,
         'Content-Type': 'text/plain'
       });
       res.end(body);
     });
     this._server.allowHalfOpen = false;
-    // maybe use a generic server.listen(options[, callback]) variant here, instead of two overloaded variants?
-    if (isDefinedAndNonNull(options, 'backlog')) {
-      this._server.listen(options.port, options.host, options.backlog, callback);
-    } else {
-      this._server.listen(options.port, options.host, callback);
-    }
-    this._closeServer = () => this._server && this._server.close();
+    this._server.listen(options.port, options.host, options.backlog, callback);
   } else if (options.server) {
     this._server = options.server;
   }
 
   if (this._server) {
-    this._onceServerListening = () => this.emit('listening');
-    this._server.once('listening', this._onceServerListening);
-    this._onServerError = (error) => this.emit('error', error);
-    this._server.on('error', this._onServerError);
-    this._onServerUpgrade = (req, socket, upgradeHead) => {
-      // copy upgradeHead to avoid retention of large slab buffers used in node core
-      var head = new Buffer(upgradeHead.length);
-      upgradeHead.copy(head);
```

Comment:
This is no longer needed. See https://github.com/jmatthewsr-ms/node-slab-memory-issues/issues/1.